Rubberduck.CodeAnalysis/Inspections/Concrete/ProcedureCanBeWrittenAsFunctionInspection.cs

```diff
@@ -36,14 +41,22 @@ protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
                 .Concat(builtinHandlers)
                 .Concat(userDeclarations.Where(item => item.IsWithEvents)));
 
+            bool HasArgumentReferencesWithIsAssignmentFlagged(QualifiedContext<ParserRuleContext> context)
+                => contextLookup.TryGetValue(context.Context.GetChild<VBAParser.ArgContext>(), out Declaration decl)
+                    ? decl.References.Any(rf => rf.IsAssignment) : false;
```

Comment:
I am not too sure whether a local function is the most readable thing here, especially an expression bodied one. Moreover, I would put each part of the ternary on a separate line. 
However, both might be personal taste. (The local function might be less surpising when not expression bodied.)